hudi-flink-datasource/hudi-flink/src/main/java/org/apache/hudi/sink/BucketStreamWriteFunction.java

```diff
@@ -66,16 +68,21 @@
   private String indexKeyFields;
 
   /**
-   * BucketID to file group mapping.
+   * BucketID should be load in this task.
    */
-  private HashMap<String, String> bucketIndex;
+  private Set<Integer> bucketToLoad;
+
+  /**
+   * BucketID to file group mapping in each partition.
+   */
+  private Map<String, Map<Integer, String>> bucketIndex;
```

Comment:
please add comments for the meaning of key and value of this map.

Reply:
fixed